include/vulkan/layer/vk_layer_settings.h

```diff
@@ -21,44 +21,23 @@
 
 #pragma once
 
-#include <string>
-#include <vector>
+#ifdef __cplusplus
+extern "C" {
+#endif 
 
-#include <vulkan/vk_layer.h>
-#include <vulkan/vulkan.h>
+#include "vk_layer_settings_ext.h"
 
-namespace vku {
+typedef void *(*VL_LAYER_SETTING_LOG_CALLBACK)(const char *pSettingName, const char *pMessage);
 
-typedef std::vector<std::string> Strings;
-typedef std::vector<std::pair<std::string, int>> List;
-typedef void *(*LAYER_SETTING_LOG_CALLBACK)(const char *setting_key, const char *message);
+// Initialize the layer settings. If 'pCallback' is set to NULL, the messages are outputed to stderr.
+void vlInitLayerSettings(const char *pLayerName, const void *pNext, VL_LAYER_SETTING_LOG_CALLBACK pCallback);
 
-// Initialize the callback function to get error messages. By default the error messages are outputed to stderr. Use nullptr to
-// return to the default behavior.
-void InitLayerSettingsLogCallback(LAYER_SETTING_LOG_CALLBACK callback);
+// Check whether a setting was set either programmatically, from vk_layer_settings.txt or an environment variable
+VkBool32 vlHasLayerSetting(const char *pSettingName);
 
-// Check whether a setting was set either from vk_layer_settings.txt or an environment variable
-bool IsLayerSetting(const char *layer_key, const char *setting_key);
+// Query setting data for a setting
+void vlGetLayerSettingData(const char *pSettingName, VkLayerSettingTypeEXT type, VkLayerSettingDataEXT *pData);
```

Comment:
Perhaps this would be a good approach to avoid having array types?

```suggestion
// Query setting data for a setting.
//
// Query once to get the number of elements.
//
// Allocate memory for the setting.
//
//Then call the function again to fill out the array of VkLayerSettingDataEXT.
void vlGetLayerSettingData(const char *pSettingName, VkLayerSettingTypeEXT type, uint32_t *pSettingCount, VkLayerSettingDataEXT *pSetting);
```

It would be similar in nature to `vkEnumeratePhysicalDevices` where you call once to get the number of devices, and then fill out the devices.

Reply:
Interresting that you are suggesting this approach because this is exactly the declaration I was working from on Friday because the array approach doesn't allow me implementing the array types.